src/wazuh_modules/wm_syscollector.c

```diff
@@ -151,6 +152,15 @@ void* wm_sys_main(wm_sys_t *sys) {
         syscollector_start_ptr = so_get_function_sym(syscollector_module, "syscollector_start");
         syscollector_stop_ptr = so_get_function_sym(syscollector_module, "syscollector_stop");
         syscollector_sync_message_ptr = so_get_function_sym(syscollector_module, "syscollector_sync_message");
+
+        void* rsync_module = NULL;
+        if(rsync_module = so_check_module_loaded("rsync"), rsync_module) {
+            void (*rsync_initialize_ptr)(full_log_fnc_t) = so_get_function_sym(rsync_module, "rsync_initialize_log_function");
+            rsync_initialize_ptr(mt_log_wrapper);
+            #ifndef WIN32
+            so_free_library(rsync_module);
+            #endif
+        }
```

Comment:
So the logging only works if the rsync_lib is loaded. What would be the case that wouldn't happen?

Reply:
Maybe I am not understanding your question. 
The `rsync_initialize_log_function` will only be called if the `rsync` module is found. 

On the other side, the `rsync_initialize_log_function` might not be available. I'm adding an extra check just in case.

Reply:
Nevermind, it was strange at first for me this dynamic loading mechanism to implement a log function, I was just not aware of this prior development.